widget/button.go

```diff
@@ -206,16 +191,15 @@ func (b *Button) CreateRenderer() fyne.WidgetRenderer {
 	objects := []fyne.CanvasObject{
 		text,
 	}
-	var shadow *shadow
-	if !b.HideShadow {
-		shadow = newShadow(shadowAround, theme.Padding()/2)
-		objects = append(objects, shadow)
+	shadowLevel := 1
+	if b.HideShadow {
```

Comment:
What if HideShadow is changed after the renderer is created, should buttonRenderer.Refresh notify baseRenderer?

Reply:
Probably yes.

Reply:
In the previous implementation the shadow could be hidden after the renderer was created but it could not be shown.
To do this right adds a lot of code without a real use case.

I think that shadows will be moved out of the renderers and into the rendering engine itself in the long run.
That's because shadows should be cast by higher elevated elements according to the material design. For this we have to introduce layers into the rendering process and then we are able to cast shadows correctly.
Until then I would leave this as it is now, with the shadow not be hideable/showable after renderer creation.

Reply:
> into the rendering engine itself

Yes yes yes

> Until then I would leave this as it is now, with the shadow not be hideable/showable after renderer creation.

Okay for now, maybe file an issue so we don't forget about it

Reply:
In theory that's OK - but it is possible that a renderer is created before widget setup is complete (I mention it out of experience). However if all the windgets that set this to true seem to be working then maybe we can leave it as-is.

Reply:
I checked all and didn't found any issues.